meta/data.go

```diff
@@ -172,6 +172,11 @@ func (data *Data) DropRetentionPolicy(database, name string) error {
 		return ErrDatabaseNotFound
 	}
 
+	// Prohibit dropping the default retention policy.
+	if di.DefaultRetentionPolicy == name {
```

Comment:
Retention policy names are already case sensitive, so this just continues with this policy. If we ever change that, we'll need to change a lot more code.